### PR TITLE
Update brew and formulae for Mojave

### PR DESCRIPTION
**Pull request by obelisk**
Brew needs to be updated for Mojave then some features are deprecated in favour of new ones. This should fix the system for 10.14.

### CODE REVIEW

**Comment by obelisk** [comment]
jenkins test codeaudit

**Comment by obelisk** [comment]
jenkins test code audit

**Comment by obelisk** [comment]
ok to test

File: tools/provision/formula/ccache.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ def install
     end
   end
 
-  def caveats; <<-EOS.undent
+  def caveats; <<~EOS
```
Comment (fmanco) [inline tools/provision/formula/ccache.rb]: Is the ~ intended here?

Comment (obelisk) [reply tools/provision/formula/ccache.rb]: Good catch

**Comment by theopolis** [comment]
@obelisk is there a reason for updating the Linux commits? Can we revert them to fix #4588? We will need to deep dive into the expected version compatibility for our gcc and the Linuxbrew-provided gmp.

**Comment by obelisk** [comment]
I believe the reason was the changes that I needed to make were unsupported in  the older version of brew so I had to pull both forward

**Pull request merged**